src/webview/services/playerSettingsService.ts

```diff
@@ -57,23 +64,149 @@ export default class PlayerSettingsService {
     this._enableSeekToPlay = value === undefined ? true : value; // true by default
   }
 
+  private _enableHpf: boolean;
+  public get enableHpf() {
+    return this._enableHpf;
+  }
+  public set enableHpf(value: boolean) {
+    const newValue = value === undefined ? false : value; // false by default
+
+    if (this._enableHpf !== newValue) {
+      this._enableHpf = newValue;
+      this.dispatchEvent(
+        new CustomEvent(EventType.PS_UPDATE_ENABLE_HPF, {
+          detail: { value: this._enableHpf },
+        }),
+      );
+    }
+  }
+
+  private _hpfFrequency: number;
+  public get hpfFrequency() {
+    return this._hpfFrequency;
+  }
+  public set hpfFrequency(value: number) {
+    const newValue = getLimitedValueInRange(
+      value,
+      PlayerSettingsService.FILTER_FREQUENCY_MIN,
+      this._sampleRate / 2,
+      PlayerSettingsService.FILTER_FREQUENCY_HPF_DEFAULT,
+    );
+
+    if (this._hpfFrequency !== newValue || value !== newValue) {
```

Comment:
While other values like as defined in AnalyzeSettings are always triggering UPDATE-type events, the newly added one has a conditional branch which seems like it could cause bugs if we forget there is different behaviour. Since it has UPDATE in its name, one could argue that your code's behavior is ideal. However, I’d like to unify the approach regardless, so I’m not keen on making an exception just for this newly added item.

Given that the frequency range to display can be set in AnalyzeSettings, why don’t we stop setting individual threshold values for the filter frequencies and use maxFrequency and minFrequency in  AnalyzeSettings? This way, we can simplify the implementation to just having checkboxes for turning HPF and LPF on or off, eliminating the need for events like PS_UPDATE_HPF_FREQUENCY. This will also prevent the issue I commented on earlier from arising.